### PR TITLE
Copy app from hoster fix

### PR DESCRIPTION
**Pull request by obino**

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/2632/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/2633/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/2637/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/2642/

**Comment by obino** [comment]
ocd run https://ocd.appscale.com:8080/job/Daily%20Build/2644/

File: AppController/djinn.rb [hunk 1]
```diff
@@ -2211,28 +2213,72 @@ def get_online_users_list(secret)
...
         Djinn.log_warn("(done_uploading) couldn't talk to zookeeper " +
-          "with #{e.message}.")
-        result = "Unknown status for #{appname}: please retry."
+          "with #{except.message}.")
       end
-    else
-      result = "The #{appname} app was not found at #{location}."
+      Kernel.sleep(SMALL_WAIT)
+    }
+    Djinn.log_warn("Fail to notify zookeeper this node hosts #{appname}.")
```
Comment (tmarballi) [inline AppController/djinn.rb]: Fail -> Failed

File: AppController/djinn.rb [hunk 2]
```diff
@@ -2211,28 +2213,72 @@ def get_online_users_list(secret)
...
+      begin
+        ZKInterface.remove_app_entry(appname, my_node.private_ip)
+        return true
+      rescue FailedZooKeeperOperationException => except
+        Djinn.log_warn("not_hosting_app: got exception talking to " +
+          "zookeeper: #{except.message}.")
+      end
+      Kernel.sleep(SMALL_WAIT)
+    }
+    Djinn.log_warn("Fail to notify zookeeper this node doesn't host #{appname}.")
```
Comment (tmarballi) [inline AppController/djinn.rb]: Fail -> Failed

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/2655/

**Pull request merged**